client/navigation/utils.ts

```diff
@@ -67,20 +72,33 @@ export const getDefaultMatchExpression = ( url ) => {
 	return '^' + path + argsExpression + hashExpression;
 };
 
+interface wcNavigation {
+	menuItems: Array< Item >;
+	rootBackLabel: string;
+	rootBackUrl: string;
+	historyPatched: boolean;
+}
+
+declare global {
+	interface Window {
+		wcNavigation: wcNavigation;
+	}
+}
+
 /**
  * Adds a listener that runs on history change.
  *
  * @param {Function} listener Listener to add on history change.
  * @return {Function} Function to remove listeners.
  */
-export const addHistoryListener = ( listener ) => {
+export const addHistoryListener = ( listener: Function ) => {
```

Comment:
Just giving some suggestions: this could be even more useful if you type the arguments to the callback. Maybe you left this unclear for now because you're not certain about the type which is all good, but you can do something like:

`listener: (someArgument: SomeType ) => void`